### PR TITLE
DEVEXP-646 First batch of Sonar fixes

### PR DESCRIPTION
**Pull request by rjrudin**
This addresses low-hanging fruit. The main batch involves the methods with high code complexity, and then there's a second batch where Sonar isn't happy about some tests apparently not performing any assertions (the tests are, Sonar just doesn't realize it). Will do 2 separate PR's for those.

### CODE REVIEW

File: src/main/java/com/marklogic/mule/extension/AbstractPagingProvider.java [hunk 1]
```diff
@@ -88,14 +90,8 @@ protected final void handlePage(DatabaseClient databaseClient, Consumer<Document
...
         if (Utilities.hasText(restTransform)) {
-            ServerTransform serverTransform = new ServerTransform(restTransform);
-            if (Utilities.hasText(restTransformParameters)) {
-                String[] parametersArray = restTransformParameters.split(restTransformParametersDelimiter);
-                for (int i = 0; i < parametersArray.length; i = i + 2) {
-                    serverTransform.addParameter(parametersArray[i], parametersArray[i + 1]);
-                }
-            }
-            queryDefinition.setResponseTransform(serverTransform);
+            queryDefinition.setResponseTransform(Utilities.makeServerTransform(restTransform, restTransformParameters,
```
Comment (rjrudin) [inline src/main/java/com/marklogic/mule/extension/AbstractPagingProvider.java:93]: Was duplicated code.

File: src/main/java/com/marklogic/mule/extension/AbstractPagingProvider.java [hunk 2]
```diff
@@ -135,4 +131,39 @@ private DocumentManager.Metadata[] buildMetadataCategories(String categories) {
         }
         return transformedCategories;
     }
+
+    private QueryDefinition buildQueryDefinitionFromParams(
```
Comment (rjrudin) [inline src/main/java/com/marklogic/mule/extension/AbstractPagingProvider.java:135]: Moved here from ReadUtil as it's only used in this class.

File: src/main/java/com/marklogic/mule/extension/ConnectionProvider.java [hunk 3]
```diff
@@ -160,10 +162,8 @@ private void configureSSL(DatabaseClientBuilder builder) {
         try {
             builder.withSSLContext(tlsContextFactory.createSslContext());
         } catch (Exception e) {
-            throw new RuntimeException(String.format(
-                "Unable to create SSL context; cause: %s",
-                tlsContextFactory.getKeyStoreConfiguration().getPath(), e.getMessage()
-            ), e);
+            String message = String.format("Unable to create SSL context; cause: %s", e.getMessage());
+            throw new ModuleException(message, ErrorType.CONNECTION_ERROR, e);
```
Comment (rjrudin) [inline src/main/java/com/marklogic/mule/extension/ConnectionProvider.java:166]: Sonar is picky about generic exceptions being used - ModuleException is Mule's preferred base class for runtime exceptions.

File: src/main/java/com/marklogic/mule/extension/Utilities.java [hunk 4]
```diff
@@ -17,18 +17,16 @@
 
 import com.marklogic.client.document.ServerTransform;
 
-class Utilities {
-    static boolean hasText(String var) {
-        return ((var != null) && !var.isEmpty());
+public interface Utilities {
+
+    static boolean hasText(String value) {
```
Comment (rjrudin) [inline src/main/java/com/marklogic/mule/extension/Utilities.java:22]: Sonar didn't like "var" as a name.

File: src/main/java/com/marklogic/mule/extension/api/QueryType.java [hunk 5]
```diff
@@ -16,8 +16,8 @@
 package com.marklogic.mule.extension.api;
 
 public enum QueryType {
-    StringQuery,
-    StructuredQuery,
-    SerializedCtsQuery,
-    CombinedQuery;
+    STRING_QUERY,
```
Comment (rjrudin) [inline src/main/java/com/marklogic/mule/extension/api/QueryType.java:19]: Sonar wants all-caps and underscores.

**Review by BillFarber — APPROVED** [review]

**Pull request merged**